corehq/apps/data_interfaces/tasks.py

```diff
@@ -110,27 +110,32 @@ def check_data_migration_in_progress(domain, last_migration_check_time):
 def run_case_update_rules_for_domain(domain, now=None):
     now = now or datetime.utcnow()
 
-    run_record = DomainCaseRuleRun.objects.create(
-        domain=domain,
-        started_on=datetime.utcnow(),
-        status=DomainCaseRuleRun.STATUS_RUNNING,
-    )
+    domain_rules = AutomaticUpdateRule.by_domain(domain, AutomaticUpdateRule.WORKFLOW_CASE_UPDATE)
+    all_rule_case_types = set(domain_rules.values_list('case_type', flat=True))
 
-    if should_use_sql_backend(domain):
-        for db in get_db_aliases_for_partitioned_query():
-            run_case_update_rules_for_domain_and_db.delay(domain, now, run_record.pk, db=db)
-    else:
-        # explicitly pass db=None so that the serial task decorator has access to db in the key generation
-        run_case_update_rules_for_domain_and_db.delay(domain, now, run_record.pk, db=None)
+    for case_type in all_rule_case_types:
+        run_record = DomainCaseRuleRun.objects.create(
+            domain=domain,
+            started_on=datetime.utcnow(),
+            status=DomainCaseRuleRun.STATUS_RUNNING,
+            case_type=case_type
+        )
+
+        if should_use_sql_backend(domain):
+            for db in get_db_aliases_for_partitioned_query():
+                run_case_update_rules_for_domain_and_db.delay(domain, now, run_record.pk, case_type, db=db)
+        else:
+            # explicitly pass db=None so that the serial task decorator has access to db in the key generation
+            run_case_update_rules_for_domain_and_db.delay(domain, now, run_record.pk, case_type, db=None)
 
 
 @serial_task(
-    '{domain}-{db}',
+    '{domain}-{case-type}-{db}',
     timeout=36 * 60 * 60,
     max_retries=0,
     queue='case_rule_queue',
 )
-def run_case_update_rules_for_domain_and_db(domain, now, run_id, db=None):
+def run_case_update_rules_for_domain_and_db(domain, now, run_id, case_type, db=None):
```

Comment:
Is this safe? I think that this might cause any currently queued tasks to error because they won't have provided `case_type`.

Reply:
yes, that's correct, currently queued tasks will error but since this is run regularly I thought it wouldn't be an issue - could be wrong though.

Reply:
Yeah. as long as its run regularly that should be fine